src/db.rs

```diff
@@ -437,6 +463,9 @@ impl DbInner {
 						match reader.next()? {
 							LogAction::BeginRecord(_) => {
 								log::debug!(target: "parity-db", "Unexpected log header");
+								// [Review] on hall those validation error handling, 
+								// should we just crash the application. I mean
+								// is it ok to clear log silently?
```

Comment:
There's an issue here indeed. If the log is detected to be invalid, we should really enact all the valid logs up to this point to recover to a consistent state.